chrome/content/zotero/xpcom/itemTreeManager.js

```diff
@@ -0,0 +1,144 @@
+/*
+    ***** BEGIN LICENSE BLOCK *****
+    
+    Copyright © 2019 Corporation for Digital Scholarship
+                     Vienna, Virginia, USA
+                     https://digitalscholar.org
+    
+    This file is part of Zotero.
+    
+    Zotero is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    
+    Zotero is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+    
+    You should have received a copy of the GNU Affero General Public License
+    along with Zotero.  If not, see <http://www.gnu.org/licenses/>.
+    
+    ***** END LICENSE BLOCK *****
+*/
+
+const { getColumns, addColumn, removeColumn, getColumn } = require("zotero/itemTreeColumns");
+
+/**
+ * @typedef {import("../itemTreeColumns.jsx").ItemTreeColumnOption} ItemTreeColumnOption
+ */
+
+class ItemTreeManager {
+    _observerAdded = false;
+    constructor() {
+    }
+
+    /** 
+     * Register a custom column. All registered columns must be valid, and must have a unique dataKey.
+     * @param {ItemTreeColumnOption | ItemTreeColumnOption[]} option
+     * @async
```

Comment:
This isn't necessary — JSDoc will [automatically detect async functions](https://jsdoc.app/tags-async.html).